src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,58 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserCredentialsException;
+import core.basesyntax.exception.InvalidUserObjectException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_EMAIL_LENGTH = 6;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
+    private static final int REQUIRED_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (validateUserCredentials(user)) {
+            return storageDao.add(user);
+        }
         return null;
     }
+
+    public boolean validateUserCredentials(User user) {
+        if (user == null) {
+            throw new InvalidUserObjectException("User object cannot be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserObjectException("User already exists");
```

Comment:
It would be more appropriate to throw an `InvalidUserCredentialsException` here instead of `InvalidUserObjectException` since the issue is related to user credentials (i.e., the login already exists).